runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptionsTest.java

```diff
@@ -199,4 +199,10 @@ public void testDefaultStagingLocationUnset() {
     thrown.expectMessage("Error constructing default value for stagingLocation");
     options.getStagingLocation();
   }
+
+  @Test
+  public void testDefaultGcpRegion() {
+    DataflowPipelineOptions options = PipelineOptionsFactory.as(DataflowPipelineOptions.class);
+    assertEquals("us-central1", options.getRegion());
```

Comment:
This test will fail on any machine where the `CLOUDSDK_COMPUTE_REGION` is defined or `gcloud config compute/region` is set.

You can test your code by instantiating the DefaultGcpRegionFactory directly and ensuring that it is factored in such a way where you can pass in the environment map directly to it and similarly for testing the process execution.

Reply:
Thanks for catching this Luke. Filed #10048 to improve this.